### ISSUE TITLE
Warning for leaving off argument names

### ISSUE THREAD

**Issue by rtfeldman** [elm-lang/elm-compiler#1665]
In a discussion some time ago, @evancz mentioned that he considers it a mistake to leave off argument names when defining toplevel named functions.

This could be codified as a compiler warning, in the same way that toplevel named functions without type annotations currently give a warning. The algorithm would be:

> If a toplevel named function has fewer named arguments than total arguments, give a warning.

Examples of functions that would trigger this warning:

```elm
negateAndRound : Float -> Int
negateAndRound =
    negate >> round


sum : List number -> number
sum =
    List.foldl (+) 0
```
